cmake/FDBComponents.cmake

```diff
@@ -12,25 +12,28 @@ endif()
 # SSL
 ################################################################################
 
-set(DISABLE_TLS OFF CACHE BOOL "Don't try to find LibreSSL and always build without TLS support")
+set(DISABLE_TLS OFF CACHE BOOL "Don't try to find OpenSSL and always build without TLS support")
 if(DISABLE_TLS)
   set(WITH_TLS OFF)
 else()
   set(OPENSSL_USE_STATIC_LIBS TRUE)
   find_package(OpenSSL)
-  if(NOT OPENSSL_FOUND)
-    set(LIBRESSL_USE_STATIC_LIBS TRUE)
-    find_package(LibreSSL)
-		if (LIBRESSL_FOUND)
-			add_library(OpenSSL::SSL ALIAS LibreSSL)
-		endif()
-  endif()
-	if(OPENSSL_FOUND OR LIBRESSL_FOUND)
-    set(WITH_TLS ON)
-    add_compile_options(-DHAVE_OPENSSL)
+  if(OPENSSL_FOUND)
+    set(CMAKE_REQUIRED_INCLUDES ${OPENSSL_INCLUDE_DIR})
+    CHECK_CXX_SOURCE_COMPILES(
+      "#include <openssl/crypto.h>
+       int main() { (void) OPENSSL_INIT_NO_ATEXIT; }" OPENSSL_HAS_NO_ATEXIT)
+    if(OPENSSL_HAS_NO_ATEXIT)
+      set(WITH_TLS ON)
+      add_compile_options(-DHAVE_OPENSSL)
+    else()
+      message(STATUS "An OpenSSL version was found, but it doesn't support OPENSSL_INIT_NO_ATEXIT - Will compile without TLS Support")
```

Comment:
I think this should be `WARNING` and not `STATUS` - otherwise this will be easily missed by a user